src/main/java/org/opentripplanner/model/TripStopTimes.java

```diff
@@ -45,15 +45,23 @@ public void replace(Trip key, Collection<StopTime> list) {
         map.replace(key, sort(list));
     }
 
-    public Map<Trip, List<StopTime>> asMap() {
-        return Collections.unmodifiableMap(map);
+    /**
+     * Return a copy of the internal map. Changes in the source are not reflected
+     * in the destination (returned Map), and visa versa.
+     * <p/>
+     * The returned map is immutable.
```

Comment:
Although the map is immutable, the values of the map will be mutable lists unless additional precautions are taken. The addAll and sort() methods could be modified to make all the value Lists immutable as well.